cpp/src/lance/io/reader.cc

```diff
@@ -347,26 +336,33 @@ ::arrow::Result<std::shared_ptr<::arrow::Array>> FileReader::GetListArray(
   auto length = params.length;
   auto start = params.offset.value();
 
-  ARROW_ASSIGN_OR_RAISE(auto offsets_arr, GetPrimitiveArray(field, batch_id, params));
+  std::shared_ptr<::arrow::Array> offsets_arr;
+  if (length.has_value()) {
+    ARROW_ASSIGN_OR_RAISE(
+        offsets_arr,
+        GetPrimitiveArray(field, batch_id, ArrayReadParams(start, length.value() + 1)));
```

Comment:
why +1 here?

Reply:
So for a N-element array, the offsets has N+1 values, i.e. `0, offset0, offset1, .... offsetN`.